packages/cli/src/lib/utils/paginator.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-await-in-loop */
+// page size ranges from 200 - 1000 seen here
+// https://devcenter.heroku.com/articles/platform-api-reference#ranges
+
+// This paginator uses status code to determine passing the Next-Range header
+import {APIClient} from '@heroku-cli/command'
+
+export async function paginateRequest(client: APIClient, url: string, pageSize = 200) {
+  let isPartial = true
+  let isFirstRequest = true
+  let nextRange: string | undefined = ''
+  let aggregatedResponseBody: any[] = []
+
+  while (isPartial) {
+    const response: any = await client.get<Array<any>>(url, {
```

Comment:
I intentionally used the `any` type here because this pagination utility should be able to take any request (be it list request or a single request) and the types per request are not the same. Open to alternatives here if possible!

Reply:
In these instances we use a generic to express the type at the time of implementation.

```
export async function paginateRequest<T = unknown>(client: APIClient, url: string, pageSize = 200): Promise<T[]>
// ...
let aggregatedResponseBody: T[] = []
//...
const response: HTTP<T[]> = await client.get<T[]>(url, { //...
// ...
nextRange = response.headers['next-range'] as string
```

Reply:
The implementation would look like this:
```
const domains = await paginateRequest<Heroku.Domain>(this.heroku, `/apps/${flags.app}/domains`, 1000)
```